rfcs/convergence/native-element-props.md

```diff
@@ -0,0 +1,310 @@
+# RFC: Customizing application of native element props and `ref`
+
+@ecraig12345 @sopranopillow @behowell
+
+## Summary
+
+Instead of always applying native props and `ref` to the root, allow setting a different "primary" slot where these props should be applied. This option should be used sparingly: for input components and possibly a few other cases (options for exact criteria discussed below).
+
+Also explicitly expose the `root` slot for all components. This will allow passing native props to the root if needed for the "special case" components discussed above, and provide consistency across the library.
+
+## Problem statement
+
+For most components, it makes sense to apply native props (and `ref`) to the root element. Currently the `root` slot always gets native element props passed to the component, and is not explicitly exposed via the API.
+
+The input components are a little unique because the actual `<input>` _cannot_ be the root element of the component, since we need various wrappers for styling (and `<input>` can't have children). However, users may expect that some or all native props passed to the root of the component would be applied to the `<input>` itself. This may also be a requirement for certain 3rd-party form libraries.
+
+For example, rendering a `<Checkbox/>` actually does this:
+
+```tsx
+<div>
+  <div ... /> {/* checkmark */}
+  <input type="checkbox" />
+  <label {...slotProps.label} />
+</div>
+```
```

Comment:
It's not fully clear _what `Checkbox` does_, can you please modify this example to be clearer?

For example (just to show where props are spreaded):

```tsx
<Checkbox aria-label="foo" type="input" />
```

```tsx
<div {...slotProps.root}> // "aria-label" goes there
  <div {...slotProps.checkmark } />
  <input type="input" /> // "type" goes to "input"
  <label {...slotProps.label} />
</div>
```

Reply:
It's a bit beside the point in this example to show where the slot props go (I'm not sure why it's shown for the label); it's just intended to show the overall structure of these components and why the `input` can't be the actual DOM root.

Reply:
Hmm actually thinking about it more, using the slot props might be an easier way to understand the slot labeling rather than using comments